migration/workspace-query-separation/src/main/java/ddf/migration/workspace/query/separation/WorkspaceQueryMigration.java

```diff
@@ -0,0 +1,205 @@
+ */
+package ddf.migration.workspace.query.separation;
+
+import static org.codice.ddf.catalog.ui.metacard.workspace.WorkspaceConstants.WORKSPACE_QUERIES;
+import static org.codice.ddf.catalog.ui.metacard.workspace.WorkspaceConstants.WORKSPACE_TAG;
+
+import com.google.common.annotations.VisibleForTesting;
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.data.types.Core;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.operation.CreateResponse;
+import ddf.catalog.operation.Query;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.UpdateResponse;
+import ddf.catalog.operation.impl.CreateRequestImpl;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.operation.impl.UpdateRequestImpl;
+import ddf.catalog.source.IngestException;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.InputTransformer;
+import ddf.catalog.util.impl.ResultIterable;
+import ddf.migration.api.DataMigratable;
+import ddf.security.Subject;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Serializable;
+import java.nio.charset.StandardCharsets;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.codice.ddf.security.common.Security;
+import org.opengis.filter.Filter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class WorkspaceQueryMigration implements DataMigratable {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(WorkspaceQueryMigration.class);
+
+  private final CatalogFramework catalogFramework;
+
+  private final FilterBuilder filterBuilder;
+
+  private final InputTransformer xmlInputTransformer;
+
+  private static final Security SECURITY = Security.getInstance();
+
+  public WorkspaceQueryMigration(
+      CatalogFramework catalogFramework,
+      FilterBuilder filterBuilder,
+      InputTransformer xmlInputTransformer) {
+    this.catalogFramework = catalogFramework;
+    this.filterBuilder = filterBuilder;
+    this.xmlInputTransformer = xmlInputTransformer;
+  }
+
+  @Override
+  public void migrate() {
+    Subject systemSubject = SECURITY.runAsAdmin(SECURITY::getSystemSubject);
+
+    if (systemSubject == null) {
+      LOGGER.debug("An error occurred while attempting to run this migration as admin");
+      return;
+    }
+
+    systemSubject.execute(this::migrateWorkspaces);
+  }
+
+  @VisibleForTesting
+  void migrateWorkspaces() {
+    LOGGER.debug("Beginning workspace query data migration");
+    System.out.println("Started: Workspace Query Data Migration");
```

Comment:
See my comment in the migration command class about printing migration task details.